pinot-common/src/main/java/org/apache/pinot/common/utils/fetcher/SegmentFetcherFactory.java

```diff
@@ -34,29 +34,40 @@
 
 
 public class SegmentFetcherFactory {
-  private SegmentFetcherFactory() {
-  }
+  private final static SegmentFetcherFactory instance = new SegmentFetcherFactory();
 
   static final String SEGMENT_FETCHER_CLASS_KEY_SUFFIX = ".class";
   private static final String PROTOCOLS_KEY = "protocols";
+  private static final String AUTH_TOKEN_KEY = "auth.token";
   private static final String ENCODED_SUFFIX = ".enc";
 
   private static final Logger LOGGER = LoggerFactory.getLogger(SegmentFetcherFactory.class);
-  private static final Map<String, SegmentFetcher> SEGMENT_FETCHER_MAP = new HashMap<>();
-  private static final SegmentFetcher DEFAULT_HTTP_SEGMENT_FETCHER = new HttpSegmentFetcher();
-  private static final SegmentFetcher DEFAULT_PINOT_FS_SEGMENT_FETCHER = new PinotFSSegmentFetcher();
-
-  static {
```

Comment:
Need to look at the history of this class carefully. The last time we made some changes like this, we had issues where the controller (I think) had an exception, probably during startup. It was when this whole thing was in com.linkedin space, so we don't have the history readily available.

Reply:
The use of this class is pretty messy. The fetchers here were instantiated statically without configuration, but we now may need to inject authTokens. I've therefore refactored the class to maintain its public (static) interface, while setting properties in the starter before use.
With my manual testing, quickstart, and integration tests things seem to work, but I bet there;s a corner case lurking somewhere.

Reply:
I examined the history dating back to PRs #1970 (caused a problem), #1973, #2171, #2200, #2210, and #2238. I am not able to spot anything that may cause problems. If the code structure will allow it,  if you can commit this class independently that would be awesome. If not, we will take this.